docs/decisions/0002-asynchronous-processing.rst

```diff
@@ -0,0 +1,28 @@
+############################
+0002 Asynchronous Processing
+############################
+
+Status
+******
+
+**Draft**
+
+Context
+*******
+
+The Commerce Coordinator is most significantly a broker for network calls to other services. This pattern can introduce latency and frailty into the entire service when a downstream service is experiencing performance or availability issues. EdX has experience mitigating these kinds of issues by using a message queue and separating the processing of the networking requests from the processing of APIs and other low-latency work using Celery queues and tasks.
+
+Decisions
+*********
+
+- This application will primarily utilize an asynchronous message queue to handle communication with other services by default.
+- This application will utilize Celery as the message broker.
+- Celery tasks will be run in the context of Celery worker processes, not in the API-providing Commerce Coordinator service.
+- Code for API endpoints, Celery tasks, and task handlers should be co-located in one place and released together.
+
```

Comment:
^ These last two bullets are a little confusing to me. Could you say a little more about what'll be suitable to go into an API endpoint vs a celery task? Or do you mean we'll have API endpoints that kick off celery tasks rather than doing the work themselves?

Reply:
Thanks for the feedback, I'll add some more info here. I meant the latter, but I can definitely clarify it.

Reply:
that's helpful- thanks!